tests/components/unifi/test_controller.py

```diff
@@ -233,47 +236,28 @@
     assert len(controller.listeners) == 0
 
 
-async def test_failed_update_failed_login(hass):
-    """Running update can handle a failed login."""
-    controller = await setup_unifi_integration(hass)
-
-    with patch.object(
-        controller.api.clients, "update", side_effect=aiounifi.LoginRequired
-    ), patch.object(controller.api, "login", side_effect=aiounifi.AiounifiException):
-        await controller.async_update()
-    await hass.async_block_till_done()
-
-    assert controller.available is False
-
-
-async def test_failed_update_successful_login(hass):
-    """Running update can login when requested."""
-    controller = await setup_unifi_integration(hass)
-
-    with patch.object(
-        controller.api.clients, "update", side_effect=aiounifi.LoginRequired
-    ), patch.object(controller.api, "login", return_value=Mock(True)):
-        await controller.async_update()
-    await hass.async_block_till_done()
-
-    assert controller.available is True
-
-
-async def test_failed_update(hass):
-    """Running update can login when requested."""
+async def test_wireless_client_event_calls_update_wireless_devices(hass):
+    """Call update_wireless_devices method when receiving wireless client event."""
     controller = await setup_unifi_integration(hass)
 
-    with patch.object(
-        controller.api.clients, "update", side_effect=aiounifi.AiounifiException
-    ):
-        await controller.async_update()
-    await hass.async_block_till_done()
-
-    assert controller.available is False
+    with patch(
+        "homeassistant.components.unifi.controller.UniFiController.update_wireless_clients",
+        return_value=None,
+    ) as wireless_clients_mock:
+        controller.api.websocket._data = {
+            "meta": {"rc": "ok", "message": "events"},
+            "data": [
+                {
+                    "datetime": "2020-01-20T19:37:04Z",
+                    "key": aiounifi.events.WIRELESS_CLIENT_CONNECTED,
+                    "msg": "User[11:22:33:44:55:66] has connected to WLAN",
+                    "time": 1579549024893,
+                }
+            ],
+        }
+        controller.api.session_handler("data")
 
-    await controller.async_update()
-    await hass.async_block_till_done()
-    assert controller.available is True
+        assert wireless_clients_mock.assert_called_once
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4925854)